src/test/container.go

```diff
@@ -117,3 +119,19 @@ func retrieveFile(target *core.BuildTarget, cid []byte, filename string, warn bo
 		}
 	}
 }
+
+// runPlzSandboxedTest runs a test using please_sandbox, which simply restricts network & process
+// namespaces. It's only available on Linux and must be installed separately, since it has to be
+// suid root in order to use CLONE_NEWNET.
+func runPlzSandboxedTest(state *core.BuildState, target *core.BuildTarget) ([]byte, error) {
```

Comment:
As discussed, ideally it would be good to be able to use the rest of the namespaces, especially the PID, UID, and filesystem to make the builds even more hermetic. Regarding the FS namespace, that would just obfuscate mounts so it shouldn't cause issues (i.e I think both of us were thinking of this + chroot earlier). I don't know if we'll need to let the user control these via the build rule, hopefully there aren't such use cases